athena-vertica/athena-vertica.yaml

```diff
@@ -0,0 +1,102 @@
+Transform: 'AWS::Serverless-2016-10-31'
+
+Metadata:
+  AWS::ServerlessRepo::Application:
+    Name: VerticaAthenaConnector
+    Description: verticaAthenaConnector Description
+    Author: 'default author'
+    SpdxLicenseId: Apache-2.0
+    LicenseUrl: LICENSE.txt
+    ReadmeUrl: README.md
+    Labels: ['athena-federation']
+    HomePageUrl: https://github.com/awslabs/aws-athena-query-federation
+    SemanticVersion: 2021.3.3
+    SourceCodeUrl: https://github.com/awslabs/aws-athena-query-federation
+
+# Parameters are CloudFormation features to pass input
+# to your template when you create a stack
+Parameters:
+  AthenaCatalogName:
+    Description: 'The name you will give to this catalog in Athena. It will also be used as the function name. This name must satisfy the pattern ^[a-z0-9-_]{1,64}$'
+    Type: String
+    AllowedPattern: ^[a-z0-9-_]{1,64}$
+    Default: "v223"
+  SpillBucket:
+    Description: 'The name of the bucket where this function can spill data.'
+    Type: String
+    Default: "test-connector-west"
+  DataBucket:
+    Description: "The bucket where this tutorial's data lives."
+    Type: String
+    Default: "test-connector-west"
+  ExportBucket:
+    Description: "The bucket where the Vertica Query results will be exported."
+    Type: String
+    Default: "test-connector-west"
+  SpillPrefix:
+    Description: 'The prefix within SpillBucket where this function can spill data.'
+    Type: String
+    Default: athena-spill
+  LambdaTimeout:
+    Description: "Maximum Lambda invocation runtime in seconds. (min 1 - 900 max)"
+    Default: 900
+    Type: Number
+  LambdaMemory:
+    Description: "Lambda memory in MB (min 128 - 3008 max)."
+    Default: 3008
+    Type: Number
+  DisableSpillEncryption:
+    Description: "WARNING: If set to 'true' encryption for spilled data is disabled."
+    Default: "false"
+    Type: String
+  VerticaConnectionString:
+    Description: 'The Vertica connection details to use by default if not catalog specific connection is defined and optionally using SecretsManager (e.g. ${secret_name}).'
+    Type: String
+    Default: "jdbc:vertica://<HOST_NAME>:<PORT_NAME>/<DB_NAME>?user=<USER>&password=<PASSWORD>"
```

Comment:
Take a look at our docdb connector, we allow customers to provide part or all of the connection string using SecretsManager. This avoids then having sensitive data in the connection string. Below is an excerpt from the docdb readme which shows how it works. Most of the code you need is already provided by MetadataHandler...it has a function will will find and replace any 'secrets' in a string with the actual value from SecretsManager.

> To support these two SQL statements we'd need to add two environment variables to our Lambda function:
> 
> docdb_instance_1 - The value should be the DocumentDB connection details in the format of:mongodb://:@:/?ssl=true&ssl_ca_certs=rds-combined-ca-bundle.pem&replicaSet=rs0
> docdb_instance_2 - The value should be the DocumentDB connection details in the format of: mongodb://:@:/?ssl=true&ssl_ca_certs=rds-combined-ca-bundle.pem&replicaSet=rs0

> You can also optionally use SecretsManager for part or all of the value for the preceeding connection details.  For example, if I set a Lambda environment variable for docdb_instance_1 to be "mongodb://${docdb_instance_1_creds}@myhostname.com:123/?ssl=true&ssl_ca_certs=rds-combined-ca-bundle.pem&replicaSet=rs0" the Athena Federation SDK will automatically attempt to retrieve a secret from AWS SecretsManager named "docdb_instance_1_creds" and inject that value in place of "${docdb_instance_1_creds}". Basically anything between ${...} is attempted as a secret in SecretsManager. If no such secret exists, the text isn't replaced.

Reply:
Used the above approach to resolve user credentials in connection string.